zio-http/src/main/scala/zhttp/service/ServerResponseWriter.scala

```diff
@@ -104,9 +105,12 @@ private[zhttp] final class ServerResponseWriter[R](
         unsafeWriteFileContent(unsafeGet())
         releaseAndRead(jReq)
 
-      case d: HttpData.UnsafeAsync =>
-        runtime.unsafeRun(ctx) {
-          writeStreamContent(d.toByteBufStream).ensuring(ZIO.succeed(releaseAndRead(jReq)))
+      case HttpData.UnsafeAsync(unsafeRun) =>
+        unsafeRun { ch => msg =>
+          ctx.writeAndFlush(msg.content).addListener { (_: Future[_]) =>
+            if (msg.isLast) ctx.writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT): Unit
+            else ch.read()
```

Comment:
I think this should also work.
```scala
ctx.writeAndFlush(msg.content)
if(!msg.isLast) ctx.read()
```

Reply:
No need for 100GB test - the code above fails all of my tests.

Reply:
This one passes current tests - I'll check it with 100GB:
```scala
        unsafeRun { ch => msg =>
          ctx.writeAndFlush(msg.content)
          if (msg.isLast) ctx.writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT): Unit
          else ch.read()
        }
```

Reply:
@tusharmath according to my tests Client calls with infinite streams (`HttpData.fromStream`)
 - never returns - there is no way to process response while sending request. It's a blocker for any use-case with infinite request/response streams.
 - leaks memory - server reads request data stream even though nobody is consuming the response data stream. It blocks any proxy-like usage. It seems like there is no back pressure.

I have these issues even with `toByteBufStream` approach.